### PR TITLE
Add more styling flexibility to chip

### PR DESCRIPTION
**Pull request by ruinunes**
Concerning component http://materializecss.com/chips.html
This adds scss variables for `height`, `font-size`, `font-weight` and `line-height`.

### CODE REVIEW

File: vendor/assets/stylesheets/materialize/_variables.scss [hunk 1]
```diff
@@ -110,6 +110,10 @@ $chip-bg-color: #e4e4e4 !default;
 $chip-border-color: #9e9e9e !default;
 $chip-selected-color: #26a69a !default;
 $chip-margin: 5px !default;
+$chip-height: 32px !default;
+$chip-line-height: 32px !default;
+$chip-font-size: 13px !default;
+$chip-font-weight: 500 !default;
```
Comment (lgavillet) [inline vendor/assets/stylesheets/materialize/_variables.scss]: This is the default library, no changes should be done here. Please move it there, if needed: app/assets/stylesheets/mtl/extend/_chips.scss

File: app/assets/stylesheets/mtl/extend/_chips.scss [hunk 2]
```diff
@@ -1,3 +1,7 @@
 .chip {
   white-space: nowrap;
+  height: $chip-height;
+  font-size: $chip-font-size;
+  font-weight: $chip-font-weight;
+  line-height: $chip-line-height;
```
Comment (lgavillet) [inline app/assets/stylesheets/mtl/extend/_chips.scss]: Possible to add those to https://github.com/at-point/mtl/blob/master/lib/generators/mtl/templates/_variables.scss with the default values?

Comment (ruinunes) [inline app/assets/stylesheets/mtl/extend/_chips.scss]: https://github.com/at-point/mtl/commit/c9579194e6c95a8c2a7f1df26e2cd467444646d6

File: lib/generators/mtl/templates/_variables.scss [hunk 3]
```diff
@@ -108,6 +108,10 @@ $chip-bg-color: #e4e4e4 !default;
 $chip-border-color: #9e9e9e !default;
 $chip-selected-color: #26a69a !default;
 $chip-margin: 5px !default;
+$chip-height: 22px !default;
+$chip-line-height: 23px !default;
+$chip-font-size: .7rem !default;
+$chip-font-weight: normal !default;
```
Comment (lgavillet) [inline lib/generators/mtl/templates/_variables.scss]: 1) This is not the default values of Materialize https://github.com/Dogfalo/materialize/blob/master/sass/components/_chips.scss
2) While checking the values, found that those values might be reported here: https://github.com/Dogfalo/materialize/blob/master/sass/components/_chips.scss#L14-L28 So, can you please update accordingly please?

Comment (ruinunes) [inline lib/generators/mtl/templates/_variables.scss]: 🐇 🔪 right. 416e263499881a2f869c48c08ef400ff1772ba7c

**Comment by ruinunes** [comment]
@lgavillet updated.

**Comment by lgavillet** [comment]
Missing point 2 (about the images and close)

**Comment by ruinunes** [comment]
Thx. I've added changes affecting child elements.
As a try, experiment with some values, e.g.:
`$chip-height` and `$chip-line-height` to `100px`
`$chip-font-size: 24px`

File: app/assets/stylesheets/mtl/extend/_chips.scss [hunk 4]
```diff
@@ -1,3 +1,20 @@
...
+  > img {
+    border-radius: $chip-border-radius;
+    height: $chip-height;
+    margin-right: $chip-height / 5;
+    width: $chip-height;
+  }
+  .close {
+    line-height: $chip-line-height;
+    font-size: $chip-font-size + 3px;
+    padding-left: $chip-height / 5;
```
Comment (lgavillet) [inline app/assets/stylesheets/mtl/extend/_chips.scss]: Divided by 4, no..?

File: app/assets/stylesheets/mtl/extend/_chips.scss [hunk 5]
```diff
@@ -1,3 +1,20 @@
...
+  height: $chip-height;
+  line-height: $chip-line-height;
+  font-size: $chip-font-size;
+  font-weight: $chip-font-weight;
   white-space: nowrap;
+  padding-right: $chip-height / 2;
+  > img {
+    border-radius: $chip-border-radius;
+    height: $chip-height;
+    margin-right: $chip-height / 5;
```
Comment (lgavillet) [inline app/assets/stylesheets/mtl/extend/_chips.scss]: Divided by 4, no..?

File: app/assets/stylesheets/mtl/extend/_chips.scss [hunk 6]
```diff
@@ -1,3 +1,20 @@
 .chip {
+  border-radius: $chip-border-radius;
+  height: $chip-height;
+  line-height: $chip-line-height;
+  font-size: $chip-font-size;
+  font-weight: $chip-font-weight;
   white-space: nowrap;
+  padding-right: $chip-height / 2;
+  > img {
```
Comment (lgavillet) [inline app/assets/stylesheets/mtl/extend/_chips.scss]: an empty line?

File: app/assets/stylesheets/mtl/extend/_chips.scss [hunk 7]
```diff
@@ -1,3 +1,20 @@
...
+    border-radius: $chip-border-radius;
+    height: $chip-height;
+    margin-right: $chip-height / 5;
+    width: $chip-height;
+  }
+  .close {
```
Comment (lgavillet) [inline app/assets/stylesheets/mtl/extend/_chips.scss]: empty line?

**Comment by lgavillet** [comment]
Also possible to add something in CHANGES.md & REFERENCE.md files? To explain the additional configuration vars of the chips?

File: app/assets/stylesheets/mtl/extend/_chips.scss [hunk 8]
```diff
@@ -1,3 +1,22 @@
...
+    border-radius: $chip-border-radius;
+    height: $chip-height;
+    margin-right: $chip-height / 4;
+    width: $chip-height;
+  }
+
+  .close {
+    font-size: $chip-font-size + 3px;
+    height: $chip-height;
+    line-height: $chip-line-height;
```
Comment (lgavillet) [inline app/assets/stylesheets/mtl/extend/_chips.scss]: here, shouldn't we add a padding-left as 1/4 of the font-size to keep the proper alignment, like we did in the img margin right? Asking.

Comment (ruinunes) [inline app/assets/stylesheets/mtl/extend/_chips.scss]: went with `padding-left: ($chip-font-size + 3px) / 2;`

File: lib/generators/mtl/templates/_variables.scss [hunk 9]
```diff
@@ -112,7 +112,7 @@ $chip-height: 32px !default;
 $chip-line-height: 32px !default;
 $chip-font-size: 13px !default;
 $chip-font-weight: 500 !default;
-$chip-border-radius: 500px !default;
+$chip-border-radius: 16px !default;
```
Comment (ruinunes) [inline lib/generators/mtl/templates/_variables.scss]: thx

**Pull request merged**